server/src/test/java/org/elasticsearch/action/search/TransportSearchActionTests.java

```diff
@@ -856,17 +855,17 @@ public void testShouldPreFilterSearchShards() {
         {
             SearchRequest searchRequest = new SearchRequest();
             assertFalse(TransportSearchAction.shouldPreFilterSearchShards(clusterState, searchRequest,
-                indices, randomIntBetween(2, 127)));
+                indices, randomIntBetween(2, 128)));
             assertFalse(TransportSearchAction.shouldPreFilterSearchShards(clusterState, searchRequest,
-                indices, randomIntBetween(127, 10000)));
+                indices, randomIntBetween(129, 10000)));
         }
         {
             SearchRequest searchRequest = new SearchRequest()
                 .source(new SearchSourceBuilder().query(QueryBuilders.rangeQuery("timestamp")));
             assertFalse(TransportSearchAction.shouldPreFilterSearchShards(clusterState, searchRequest,
-                indices, randomIntBetween(2, 127)));
+                indices, randomIntBetween(2, 128)));
             assertTrue(TransportSearchAction.shouldPreFilterSearchShards(clusterState, searchRequest,
-                indices, randomIntBetween(127, 10000)));
+                indices, randomIntBetween(129, 10000)));
```

Comment:
I'm not exactly sure whether this should be 129 or 128, the way I read the docs I would understand them to _not_ run pre-filter phase if <=128, but I might be wrong here. Please let me know what you think is right.

Reply:
looking at the code it would seem 129, but I would try to run the test with a static value just to be 200% sure.

Reply:
That I already did, I was just wondering what the expectations are of the code should be changed to "<=" in the appropriate place. Thanks.